packages/project-access/src/project/cap.ts

```diff
@@ -557,29 +557,38 @@ async function readPackageNameForFolder(baseUri: string, relativeUri: string): P
     return packageName;
 }
 
-let globalCdsPathCache: string;
+// Cache for request to load global cds. Cache the promise to avoid starting multiple identical requests in parallel.
+let globalCdsModulePromise: Promise<CdsFacade> | undefined;
 
 /**
  * Try to load global installation of @sap/cds, usually child of @sap/cds-dk.
  *
  * @returns - module @sap/cds from global installed @sap/cds-dk
  */
-async function loadGlobalCdsModule<T>(): Promise<T> {
-    if (!globalCdsPathCache) {
-        const versions = await getCdsVersionInfo();
-        if (!versions.home) {
-            throw Error('Can not find global installation of module @sap/cds, which should be part of @sap/cds-dk');
-        }
-        globalCdsPathCache = versions.home;
-    }
-    return loadModuleFromProject<T>(globalCdsPathCache, '@sap/cds');
+async function loadGlobalCdsModule(): Promise<CdsFacade> {
+    globalCdsModulePromise =
+        globalCdsModulePromise ??
+        new Promise<CdsFacade>((resolve, reject) => {
+            return getCdsVersionInfo().then((versions) => {
+                if (versions.home) {
+                    resolve(loadModuleFromProject<CdsFacade>(versions.home, '@sap/cds'));
+                } else {
+                    reject(
+                        new Error(
+                            'Can not find global installation of module @sap/cds, which should be part of @sap/cds-dk'
+                        )
+                    );
+                }
+            }, reject);
+        });
```

Comment:
Would the async/await notation not be more readable, what do you think?

```suggestion
            const versions = await getCdsVersionInfo();
            if (versions.home) {
                resolve(loadModuleFromProject<CdsFacade>(versions.home, '@sap/cds'));
            } else {
                reject(
                    new Error(
                        'Can not find global installation of module @sap/cds, which should be part of @sap/cds-dk'
                    )
                );
            }
```